src/scheduler/misc.c

```diff
@@ -889,43 +888,45 @@ is_num(char *str)
 	int i;
 	char c;
 	int colen_count = 0;
+	int str_len = -1;
 
 	if (str == NULL)
 		return -1;
 
 	if (str[0] == '-' || str[0] == '+')
 		str++;
 
-	for (i = 0; i < strlen(str) && (isdigit(str[i]) || str[i] == ':'); i++) {
+	str_len = strlen(str);
+	for (i = 0; i < str_len && (isdigit(str[i]) || str[i] == ':'); i++) {
 		if (str[i] == ':')
 			colen_count++;
```

Comment:
It's "colon", not "colen". I know you didn't introduce it, but there should be misspellings like this.